lldb/source/Plugins/ExpressionParser/Swift/SwiftPersistentExpressionState.cpp

```diff
@@ -226,3 +226,12 @@ bool SwiftPersistentExpressionState::GetSwiftPersistentDecls(
   return m_swift_persistent_decls.FindMatchingDecls(name, excluding_equivalents,
                                                     matches);
 }
+
+ConstString
+SwiftPersistentExpressionState::GetNextPersistentVariableName(bool is_error) {
+  llvm::SmallString<64> name;
+  llvm::raw_svector_ostream os(name);
+  os << GetPersistentVariablePrefix(is_error)
+     << m_next_persistent_variable_id++;
```

Comment:
This isn't right.  You have to switch on is_error and either use the m_next_persistent_variable_id or the m_next_persistent_error_id.  Something like:

```+ConstString 
+SwiftPersistentExpressionState::GetNextPersistentVariableName(bool is_error)
+{
+  llvm::SmallString<64> name;
+  {
+    llvm::raw_svector_ostream os(name);
+    uint32_t variable_num = is_error ? m_next_persistent_error_id++ 
+                                    : m_next_persistent_variable_id++;
+    os << GetPersistentVariablePrefix(is_error)
+       << variable_num;
+  }
+  return ConstString(name);
+}
+
```

Reply:
Fixed